Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -90,6 +128,46 @@ class APIClient {
             completion: completion
         )
     }
+    
+    /// Queues a failed request for executing later or failing after a timeout
+    func handleTokenExpirationError<Response: Decodable>(
+        urlRequest: URLRequest,
+        timeout: TimeInterval,
+        completion: @escaping (Result<Response, Error>) -> Void
+    ) {
+        requestsAccessQueue.async {
```

Comment:
do we need [weak self] here? what if the APIClient gets `deinit` while some stuff is queued?

Reply:
Imho we should keep everything alive until we complete all the requests, because we didn't show errors for instance for messages sent, so a user might've left and then these messages will never actually reach their destination. Of course keeping references to the client won't prevent a user from killing the app, but it's still better than nothing